### PR TITLE
is210-week-10-synthesizing

### PR DESCRIPTION
**Pull request by xiangyq**

### CODE REVIEW

**Comment by xiangyq** [comment]
@cheuschober 
I don't understand the error. 'customer_id'--key error, which means this key is not in the dictionary. But it should, right? Otherwise, how can we join the tables by this key?
I am confused. Please help me!

File: task_01.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        >>> sum_orders(ORDERS, CUSTOMERS)
+        {2: {'email': 'email@one.com', 'total': 20,
+        'orders': 2, 'name': 'person one'},
+        3: {'email': 'email@two.com', 'total': 15,
+        'orders': 1, 'name': 'person two'}}
+    """
+    order_data = change_orders(orders)
+    for key in order_data:
+        order_data[key].update(customers[key])
+    return order_data
```
Comment (cheuschober) [inline task_01.py]: @xiangyq, The trick, in this case, is looping the right object, first. You actually only need about 10 lines in which to do it. Notably, we're not *transforming* our primary dataset, we're returning a new one. If that's the case you're right to think about looping orders first but you're overthinking it. Create a new dataset you want to return, loop orders, and fill it as you go along.

**Comment by xiangyq** [comment]
@cheuschober 
I feel good this time. However, I still failed the test. So, I calculated the 'total' numbers by using calculator and the numbers I got by my program are the same by calculator. And then I looked at your test_task_01.py, and I noticed that total numbers you got are different. Is there any chance you got the wrong numbers for some reason?
Please take a look at my code again. I am a little desperate right now.

**Comment by cheuschober** [comment]
@xiangyq, I'll look at lunch. If there's a typo in the test then there's also a typo in my reference implementation.

**Pull request merged**